src/Symfony/Bundle/FrameworkBundle/Secrets/BaseSecretsHandler.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Symfony\Bundle\FrameworkBundle\Secrets;
+
+use Symfony\Bundle\FrameworkBundle\Secrets\Exception\SecretsMissingException;
+use Symfony\Bundle\FrameworkBundle\Secrets\Exception\InvalidSecretsFormatException;
+
+class BaseSecretsHandler
+{
+    const ENCRYPTION_METHOD = "aes-256-cbc";
+    const CIPHERTEXT_KEY = "ciphertext";
+    const IV_KEY = "iv";
+
+    protected $encryptedSecretsLocation;
+    protected $plaintextSecretsLocation;
```

Comment:
If we can avoid a base class, that would actually be ideal. But if we are going to use a base class, let's make this private, and access it through a protected method. Much easier if we need to add deprecations later.

Reply:
do you think this would be a good candidate for a `trait SecretsHandlerTrait` instead of a base class? keeping `readSecrets` and `decryptSecretValue` here and then having things can access the constants through `SecretsWriter`